src/Autocomplete/src/Doctrine/EntitySearchUtil.php

```diff
@@ -75,9 +75,8 @@ public function addSearchClause(QueryBuilder $queryBuilder, string $query, strin
                     }
 
                     if ($i < $numAssociatedProperties - 2) {
-                        $propertyMetadata = $associatedEntityDto->getPropertyMetadata($associatedPropertyName);
-                        $targetEntity = $propertyMetadata['targetEntity'];
-                        $associatedEntityDto = $this->metadataFactory->create($targetEntity);
+                        $propertyMetadata = $associatedEntityDto->getAssociationMetadata($associatedPropertyName);
```

Comment:
Not sure if I have to add a `$associatedEntityDto->isAssociation($propertyName)` check here and use the corresponding (introduced) method.

Reply:
Not sure - this area is complex. If I understand correctly, we already know we're dealing with an association, so it should be safe as it is.

Reply:
That was my thought too, thanks!